src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -326,15 +326,23 @@ trait Infer extends Checkable {
         (samFun ne NoType) && isCompatible(tp, samFun)
       }
 
-      val tp1 = methodToExpressionTp(tp)
+      // can only compare if both types are repeated or neither is (T* is not actually a first-class type, even though it has a BTS and thus participates in subtyping)
+      (!isRepeatedParamType(tp) || isRepeatedParamType(pt)) && {
```

Comment:
We'll proceed through this if we have a Java and Scala repeated param type. `Array[T] <=> Seq[T]` adaptations might then come into play below. I haven't thought this through enough to know if this matters, but did you consider more fine grained checking ("are both/neither java repeated?" "are both/neither Scala repeated").

Reply:
I didn't give it that much thought, but I think it's ok. This is to shield us from comparing the second-class repeated param types to first-class expression types.

If `tp` is a repeated param type, so must `pt` be. If we don't check that, we'll go up the BTS of those types, hitting Seq / Array, and then relating those repeated param types to regular expression types. We should really think of repeated param types as abstract types that you can only convert out of with a conversion (that conversion is already in adapt because of the tighter conformance implemented here.)